digital_comms/runner.py

```diff
@@ -109,8 +109,57 @@ def read_parameters():
     with open(path, 'r') as ymlfile:
         for data in yaml.load_all(ymlfile):
             parameters = data['parameters']
-            for name, param in parameters.items():
-                parameters[name] = param['default_value']
+            for param in parameters:
```

Comment:
This also seems to be duplicate code, and you could probably simplify with something like:
```python
for name, param in parameters.items():
    parameters[name] = param['default_value']
```